File: algorithms/17_matlab/sorting/bubble_sort.m
function arr = bubble_sort(arr)
% BUBBLE_SORT Sorts an array using standard bubble sort algorithm
    n = length(arr);
    for i = 1:(n - 1)
        for j = 1:(n - i)
            if arr(j) > arr(j + 1)
                temp = arr(j);
                arr(j) = arr(j + 1);
                arr(j + 1) = temp;
            end
        end
    end
end

File: algorithms/08_r/sorting/bubble_sort.R
bubble_sort <- function(arr) {
  n <- length(arr)
  if (n <= 1) return(arr)
  for (i in 1:(n - 1)) {
    for (j in 1:(n - i)) {
      if (arr[j] > arr[j + 1]) {
        temp <- arr[j]
        arr[j] <- arr[j + 1]
        arr[j + 1] <- temp
      }
    }
  }
  return(arr)
}
